web/pdf_attachment_viewer.js

```diff
@@ -51,16 +51,24 @@ var PDFAttachmentViewer = (function PDFAttachmentViewerClosure() {
     this.container = options.container;
     this.eventBus = options.eventBus;
     this.downloadManager = options.downloadManager;
+
+    this._renderedCapability = pdfjsLib.createPromiseCapability();
+    this.eventBus.on('fileattachmentannotation',
+      this._appendAttachment.bind(this));
   }
 
   PDFAttachmentViewer.prototype = {
-    reset: function PDFAttachmentViewer_reset() {
+    reset: function PDFAttachmentViewer_reset(keepRenderedCapability) {
       this.attachments = null;
 
       var container = this.container;
       while (container.firstChild) {
         container.removeChild(container.firstChild);
       }
+
+      if (!keepRenderedCapability) {
```

Comment:
Could you elaborate a bit on why we need the `keepRenderedCapability` option? It's not entirely clear to me what its use is.